### DISCUSSION TITLE
Meetup

### DISCUSSION THREAD

**Discussion by gitToad** [eidolonFIRE/xcNav_web-DEPRECATED-#15] [Ideas]
Here are the general entities we are dealing with (which will be reflected in both the API and DB schema)

**Pilots**
basic pilot info incl login, contact info, preferences

**Flights**
each flight of each pilot: which meetup it was part of, duration,  take off time, landing time & location, self grading of landing etc.

**Telemetry**
periodic timestamped telemetry packets for each pilots flights: Timestamp, flightID, lat, lng, alt, hdg, spd, fuel

**Messages**
messages exchanged between pilots of a meetup during flight.   This probably should include ADSB alerts also although they  May have some additional metadata such as plane telemetry And recent path

**Overlays**	
index of available flightPlan and airspace overlays  These are uploaded by users and available to all  The DB only holds the metadata, the actual json file with the   Icon, marker, polyline, polygon data is stored in the file System as are icon png’s if any.

**Meetup** 
Defines a subset of pilots that fly together during a ...."meetup"  the pilots in this subset will get each others messages, see  each other on the map etc.

The meetup concept seems simple but needs some more discussion around how and when it is created, shared to other pilots in the meetup and ended. Each pilot can only be in one meetup at a time.

The meetup would consist of an airspace and flight plan overlay selection, a date, maybe a name, and the pilots that participate. 

Some thoughts on how to begin a meetup from Matt: one person could start the meetup by selecting the appropriate airspace and flightplan and then select a "share meetup" menu item. That would create a QR code instantly displayed on the phone which you could hold up to other pilots at the same LZ. They go to the "Scan meetup" function in their app and boom they are "in the same meetup". This is currently prototyped in the app.

But what if you want to share the meetup the week before the flight when you are not physically together. By email ?
What if a pilot wants to opt out of meetup (cant make it that day etc.)
Does a meetup allow multiple flights per pilot - as in a paracamp adventure ?
What if I just want to fly solo but still record my telemetry etc. ie a single pilot meetup ?
How does a meetup "end" ? Who says so ?

The rest of the objects are pretty straightforward, but the meetup needs more discussion I think

**Comment by eidolonFIRE** [comment]
Some thoughts on **Meetup** concept.

A problem I see with the classic "share meetup" scheme is that if you want to join the group in flight or while everybody is setting up and launching, but you don't already have the invite, is that it requires the host to take intentional steps to include you. What if you aren't that familiar with the host, or it's a big group and you aren't sure who is hosting?

Alternate idea for **Meetup**:
Meetups are organic globs. All you must do is join any member in your contact list that is in the target group you want to join. You will then added to the blob. 

Then, the only delicate bit to solve is when two blobs form when there should be one big one. Perhaps if you join someone and then they join a different blob, you are presented with the option to follow them to the new blob?

Splitting blobs down the middle might also prove a bit tedious. Perhaps it could be done spatially? As two groups diverge in flight, you could be presented to split off and others presented with option to join you?
